drivers/flash/nrf_qspi_nor.c

```diff
@@ -381,22 +385,41 @@ static int qspi_send_cmd(const struct device *dev, const struct qspi_cmd *cmd,
 }
 
 /* RDSR wrapper.  Negative value is error. */
-static int qspi_rdsr(const struct device *dev)
+static int qspi_rdsrX(const struct device *dev, uint8_t sr_num)
 {
-	uint8_t sr = -1;
+	uint8_t opcode = SPI_NOR_CMD_RDSR;
+
+	if (sr_num > 2 || sr_num == 0) {
+		return -EINVAL;
+	}
+	if (sr_num == 2) {
+		opcode = SPI_NOR_CMD_RDSR2;
+	}
+	uint8_t sr = 0xFF;
 	const struct qspi_buf sr_buf = {
 		.buf = &sr,
 		.len = sizeof(sr),
 	};
 	struct qspi_cmd cmd = {
-		.op_code = SPI_NOR_CMD_RDSR,
+		.op_code = opcode,
 		.rx_buf = &sr_buf,
 	};
 	int ret = qspi_send_cmd(dev, &cmd, false);
 
 	return (ret < 0) ? ret : sr;
 }
 
+/* RDSR wrapper.  Negative value is error. */
+static inline int qspi_rdsr(const struct device *dev)
+{
+	return qspi_rdsrX(dev, 1);
+}
+
+static inline int qspi_rdsr2(const struct device *dev)
+{
+	return qspi_rdsrX(dev, 2);
+}
```

Comment:
why do we need these wrappers? Can't we just have `qspi_rdsr` that takes sr_num?

Reply:
Agreed